lib/reporter/index.js

```diff
@@ -162,6 +169,52 @@ class Reporter extends SimplifiedReporter {
     this.testResults.logAssertion(result);
   }
 
+  /**
+   * @param {TheeNode} node
+   * @param {Object} result
+   */
+  logCommandResult({node, result}) {
+
+    const isSuccess =  result === undefined ||
+      result == null ||
+      !((result instanceof Error) ||
+        (result.error instanceof Error) ||
+        result.status === -1
+      );
+
+    if (this.shouldLogCommand(node)) {
+      this.testResults.logCommand({
+        name: node.fullName,
+        args: node.args,
+        startTime: node.startTime,
+        endTime: node.endTime,
+        elapsedTime: node.elapsedTime,
+        status: isSuccess ? TestResults.TEST_PASS : TestResults.TEST_FAIL,
+        result
+      });
+    }
+  }
+
+  shouldLogCommand(node) {
+    if (!node.parent) { //root node
+      return false;
+    }
+    
+    if (node.parent.isRootNode) {
+      return true;
+    }
+
+    if (node.parent.context && node.parent.context.isUserDefined) {
+      return false;
+    }
+
+    if (node.addedInsideCallback) {
+      return true;
+    }
```

Comment:
shouldn't the condition for checking if the node is called from the callback precede the one where the check for a user-defined function is done? A callback can be from a user-defined function.

Reply:
Yes correct, updating.